xwiki-platform-core/xwiki-platform-icon/xwiki-platform-icon-api/src/main/java/org/xwiki/icon/IconRenderer.java

```diff
@@ -41,10 +42,41 @@
 
     /**
      * Generate the HTML code to display an icon.
+     *
      * @param iconName name of the icon to render
      * @param iconSet icon set that contains the icon to display
      * @return the HTML code that displays the icon or an empty string if the icon does not exist
      * @throws IconException if problems occur
      */
     String renderHTML(String iconName, IconSet iconSet) throws IconException;
+
+    /**
+     * Render an icon with the specified renderer.
+     *
+     * <p>The renderer contains velocity code which should use the <b>$icon</b> variable.
+     * For instance: <pre>String renderer = "fa fa-$icon";</pre>
+     *
+     * @param iconName name of the icon to render
+     * @param iconSet icon set that contains the icon to display
+     * @param renderer velocity code to render
+     * @return rendered icon using the renderer through velocity
+     * @throws IconException if problems occur
+     * @since 10.6RC1
+     */
+    default String renderIcon(String iconName, IconSet iconSet, String renderer) throws IconException
```

Comment:
I don't like that much the name of this method but I don't have a better name ATM. Also, I feel that we're exposing some internal method, but I guess we have to.

Reply:
@mflorea I agree that we should be careful when making internal methods public. I just wanted to avoid making a copy-paste of the method inside the `getMetaData` method. Moreover we already have the `render` and `renderHTML` methods that behave the same.

We can also name this method just `render(String iconName, IconSet iconSet, String renderer)` WDYT? That way we will have `render(String iconName, IconSet iconSet)` and `renderHTML(String iconName, IconSet iconSet)` just calling `render(String iconName, IconSet iconSet, String renderer)`.